code/iaas/logic/src/main/java/io/cattle/platform/process/containerevent/ContainerEventPreCreate.java

```diff
@@ -0,0 +1,69 @@
+package io.cattle.platform.process.containerevent;
+
+import static io.cattle.platform.core.model.tables.AgentTable.*;
+import static io.cattle.platform.core.model.tables.ContainerEventTable.*;
+import io.cattle.platform.core.constants.AgentConstants;
+import io.cattle.platform.core.model.Agent;
+import io.cattle.platform.core.model.ContainerEvent;
+import io.cattle.platform.core.model.Host;
+import io.cattle.platform.engine.handler.HandlerResult;
+import io.cattle.platform.engine.handler.ProcessPreListener;
+import io.cattle.platform.engine.process.ProcessInstance;
+import io.cattle.platform.engine.process.ProcessState;
+import io.cattle.platform.object.util.DataAccessor;
+import io.cattle.platform.process.common.handler.AbstractObjectProcessLogic;
+import io.cattle.platform.util.type.Priority;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.inject.Named;
+
+import org.jooq.exception.InvalidResultException;
+
+@Named
+public class ContainerEventPreCreate extends AbstractObjectProcessLogic implements ProcessPreListener, Priority {
+
+    @Override
+    public String[] getProcessNames() {
+        return new String[] { "containerevent.create" };
+    }
+
+    @Override
+    public HandlerResult handle(ProcessState state, ProcessInstance process) {
+        // event's account id is set to the agent that submitted. This will
+        // change it to the actual user's account id.
+        // Checks to make sure agent's resource account id and the host's
+        // account id match
+        ContainerEvent event = (ContainerEvent)state.getResource();
+        Agent agent = null;
+        try {
+            agent = objectManager.findOne(Agent.class, AGENT.ACCOUNT_ID, event.getAccountId());
+        } catch (InvalidResultException e) {
+            // Found more than one agent, don't do anything.
```

Comment:
I guess another Exception I'm not sure whether we want to bubble up or not (or possibly log)?